instruqt-tracks/vault-auth-aws/track.yml

```diff
@@ -0,0 +1,309 @@
+      header_value=instruqt.vault.example.com \
+      role=my-role-iam
+    ```
+
+    Note that above we specify `header_value` and `role`, but we did not
+    explicitly pass any credentials to the login command.
+
+    That is because the Vault client will look for credentials environment
+    variables then `~/.aws/credentials`, IAM instance profile, or ECS task role,
+    in that order. In this case, you didn't have to enter anything because
+    `$AWS_ACCESS_KEY_ID` and `$AWS_SECRET_ACCESS_KEY` are already set in the
+    environment.
+
+    You can always specify the credentials `aws_access_key_id`,
+    `aws_secret_access_key` and `aws_security_token` as arguments of the `login`
+    command, but this is not recommended.
+
+    Take the `token` value that you received in the response from Vault and
+    paste it into the text field in the Vault UI tab, using the Token method. You
+    should be able to log into the Vault UI.
+
+    Move back to the CLI, and confirm that you are no longer logged in with the root
+    token by running the following command.
+
+    ```
+    vault auth list
+    ```
+
+    Now that you are no longer using the root token, you cannot list the auth
+    methods in Vault, as you no longer have the right permissions.
+  notes:
+  - type: text
+    contents: |-
+      The `iam` auth method authenticates AWS IAM principals. This can include IAM
+      users, IAM roles assumed from other accounts, AWS Lambdas that are
+      launched in an IAM role, or even EC2 instances that are launched in an
+      IAM instance profile.
+
+      The `iam` auth method authenticates by having clients provide a specially
+      signed AWS API request which the method then passes on to AWS to validate
+      the signature and tell Vault who created it. Upon successful authentication,
+      a Vault token will be returned to you.
+  - type: text
+    contents: |-
+      The actual secret (i.e., the AWS secret access key) is never transmitted
+      over the wire, and the AWS signature algorithm automatically expires
+      requests after 15 minutes, providing simple and robust protection against
+      replay attacks.
+  tabs:
+  - title: Log In Using the AWS Auth Method
+    type: terminal
+    hostname: vault-server
+  - title: Vault UI
+    type: service
+    hostname: vault-server
+    port: 8200
+  difficulty: basic
+  timelimit: 300
+- slug: logout-aws-auth-method
+  id: 6xmfvpxd2gmd
+  type: challenge
+  title: Log Out of Vault
+  teaser: Remove your cached token and demonstrate loss of access to Vault.
+  assignment: |-
+    Now you are ready to perform the logout operation, by removing your
+    cached Vault token.
+
+    ```
+    rm ~/.vault-token
+    ```
+
+    With your cached Vault token removed, you will no longer be authenticated to Vault.
+    Confirm this by trying to list all the auth methods one more time.
+
+    ```
+    vault auth list
+    ```
+
+    You should see a missing client token error.
```

Comment:
I would add a comment after this pointing out that the user could export their root token (from /root/token) if they wanted to be able to run more Vault CLI commands.  There is no need to run more commands in the track, but in the real world after testing logging in with the AWS auth method, a user would want to re-export their root token.